### PR TITLE
Add arithmetic operators to valarray #381

### PR DESCRIPTION
**Pull request by archangel4**
overloaded + operator , let me know if this works , ill overload - operator too .
also please give me some feedback , this is my first pull request ever and first time contributing to an open source project .
thanks

### CODE REVIEW

**Comment by kylelutz** [comment]
So this code doesn't compile, see my comment on issue #381 for some example code.

Also, take a look at the [Developer's Guide](https://github.com/kylelutz/compute/wiki/Developer%27s-Guide) and [CONTRIBUTING.md](https://github.com/kylelutz/compute/blob/master/CONTRIBUTING.md) file for some instructions on developing and submitting your patch. Thanks!

**Comment by archangel4** [comment]
I can't seem to get the code to compile , help.

**Comment by kylelutz** [comment]
Did you try the code I provided in issue #381? That code should compile fine. What errors are you getting?

**Comment by archangel4** [comment]
Yes I used that exact same code , the Travis ci build fails every time.

**Comment by kylelutz** [comment]
Oops, I forgot to add the result range for `transform()`, try this:
```
valarray<T> operator+(const valarray<T> &other) const
{
    const context &context = m_buffer.get_context();
    command_queue queue(context, context.get_device());
    valarray<T> result(size(), context);
    transform(begin(), end(), other.begin(), result.begin(), plus<T>(), queue);
    queue.finish();
    return result;
}
```

File: include/boost/compute/container/valarray.hpp [hunk 1]
```diff
@@ -90,7 +90,30 @@ class valarray
     ~valarray()
     {
     }
-
+ 
+    valarray<T> operator+(const valarray<T> &rhs) const
++    {
```
Comment (roshanr95) [inline include/boost/compute/container/valarray.hpp]: Extra '+' symbol seems to be the problem... Same thing in a few more lines that follow...

**Pull request closed without merging**